src/applications/mhv/secure-messaging/components/shared/AlertBackgroundBox.jsx

```diff
@@ -45,6 +71,38 @@ const AlertBackgroundBox = props => {
     dispatch(focusOutAlert());
   };
 
+  const location = useLocation();
+  const lastPathName = formatPathName(location.pathname, 'Messages');
+
+  // sets custom server error messages for the landing page and folder view pages
+  useEffect(
+    () => {
+      let content = activeAlert?.content;
+      const isServiceOutage = activeAlert?.response?.code === SERVICE_OUTAGE;
+      const isErrorAlert = activeAlert?.alertType === 'error';
+      if (!props.isLandingPage && (isServiceOutage || isErrorAlert)) {
+        content = SERVER_ERROR_503;
+      }
+      setAlertContent(content);
+    },
+    [SERVER_ERROR_503, SERVICE_OUTAGE, activeAlert, props.isLandingPage],
+  );
+
+  useInterval(() => {
+    const shouldRetrieveFolders =
+      activeAlert?.response?.code === SERVICE_OUTAGE ||
+      folder?.folderId === undefined;
+
+    if (shouldRetrieveFolders) {
+      dispatch(retrieveFolder(folder?.folderId));

```

Comment:
Would you please help me understand the attempt to retrieve a list of folders when we have a know service outage?

Reply:
I guess I see why we would want to reattempt in 30 seconds, but is it a business requirement?